mne/bem.py

```diff
@@ -0,0 +1,321 @@
+    for p in range(u['nfit'] - 1):
+        dot = np.dot(u['uu'][p], u['y'])
+        for k in range(u['nterms'] - 1):
+            u['resi'][k] = u['resi'][k] - u['uu'][p, k] * dot
+
+    # Return their sum of squares
+    return np.dot(u['resi'], u['resi'])
+
+
+def _fwd_eeg_fit_berg_scherg(m, nterms, nfit):
+    """Fit the Berg-Scherg equivalent spherical model dipole parameters"""
+    assert nfit >= 2
+    u = dict(y=np.zeros(nterms-1), resi=np.zeros(nterms-1),
+             nfit=nfit, nterms=nterms, M=np.zeros((nterms - 1, nfit - 1)))
+
+    # (1) Calculate the coefficients of the true expansion
+    u['fn'] = _fwd_eeg_get_multi_sphere_model_coeffs(m, nterms + 1)
+
+    # (2) Calculate the weighting
+    f = (min([layer['rad'] for layer in m['layers']]) /
+         max([layer['rad'] for layer in m['layers']]))
+
+    # correct weighting
+    k = np.arange(1, nterms + 1)
+    u['w'] = np.sqrt((2.0 * k + 1) * (3.0 * k + 1.0) /
+                     k) * np.power(f, (k - 1.0))
+    u['w'][-1] = 0
+
+    # Do the nonlinear minimization, constraining mu to the interval [-1, +1]
+    mu_0 = np.random.RandomState(0).rand(nfit) * f
+    fun = partial(_one_step, u=u)
+    cons = []
+    for ii in range(nfit):
+        for val in [1., -1.]:
+            cons.append({'type': 'ineq',
+                         'fun': lambda x: np.array([val * x[ii] + 1.]),
+                         'jac': lambda x: np.array([0.] * ii + [val] +
+                                                   [0.] * (nfit - ii - 1))})
+    mu = optimize.minimize(fun, mu_0, constraints=cons, method='COBYLA',
+                           tol=1e-2).x
+
+    # (6) Do the final step: calculation of the linear parameters
+    rv, lambda_ = _compute_linear_parameters(mu, u)
+    order = np.argsort(mu)[::-1]
+    mu, lambda_ = mu[order], lambda_[order]  # sort: largest mu first
+
+    m['mu'] = mu
+    # This division takes into account the actual conductivities
+    m['lambda'] = lambda_ / m['layers'][-1]['sigma']
+    m['nfit'] = nfit
+    return rv
+
+
+@verbose
+def make_sphere_model(r0=(0., 0., 0.04), head_radius=0.09, info=None,
+                      verbose=None):
+    """Create a spherical model for forward solution calculation
+
+    Parameters
+    ----------
+    r0 : array-like | str
+        Head center to use (in head coordinates). If 'auto', the head
+        center will be calculated from the digitization points in info.
+    head_radius : float | str | None
+        If float, compute spherical shells for EEG using the given radius.
+        If 'auto', estimate an approriate radius from the dig points in Info,
+        If None, exclude shells.
+    info : instance of mne.io.meas_info.Info | None
+        Measurement info. Only needed if ``r0`` or ``head_radius`` are
+        ``'auto'``.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    bem : dict
+        A spherical BEM.
```

Comment:
I don't like the BEM naming for an analytical sphere solution

is it called BEM is the C code?